app/controllers/settings/password.js

```diff
@@ -0,0 +1,34 @@
+import Ember from 'ember';
+
+const { Controller } = Ember;
+
+export default Controller.extend({
+  actions: {
+    changePassword(passwordData) {
+      this.set('isLoading', true);
+      let payload = {
+        'data': {
+          'old-password' : passwordData.passwordCurrent,
+          'new-password' : passwordData.passwordNew
+        }
+      };
+      this.get('loader')
+        .post('/auth/change-password', payload)
+        .then(data => {
+          if (data.error) {
```

Comment:
If the server returns a Http status code anything other than 2xx, the response will come in catch. The server needs to set a 2xx status code only for a success. Any error should be a 4xx code. And response will be in catch. Please check it and open a server issue if server is wrong. And please move error logic to catch.